controllers/volumereplication_controller.go

```diff
@@ -217,7 +217,9 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 			replicationErr = r.markVolumeAsSecondary(instance, logger, volumeHandle, parameters, secret)
 			if replicationErr == nil {
 				logger.Info("volume is not ready to use")
-				err = r.updateReplicationStatus(instance, logger, getCurrentReplicationState(instance), "volume is degraded")
+				// set the status.State to secondary as the
+				// instance.Status.State is primary for the first time.
+				err = r.updateReplicationStatus(instance, logger, getReplicationState(instance), "volume is marked secondary and is degraded")
```

Comment:
We can always pass SecondaryState here since we want to mark it as secondary. 

```suggestion
				err = r.updateReplicationStatus(instance, logger, replicationv1alpha1.SecondaryState, "volume is marked secondary and is degraded")
```